aws-cpp-sdk-core/source/utils/threading/Executor.cpp

```diff
@@ -80,6 +84,7 @@ bool PooledThreadExecutor::SubmitToThread(std::function<void()>&& fn)
         m_tasks.push(fnCpy);
     }
 
+	std::unique_lock<std::mutex> locker(m_syncPointLock);

```

Comment:
The background is Executor create many threads to fetch and run task from taskQueue of Executor.
Also Executor is responsible for set up the threads' status: continue or not.
Check out ThreadTask::MainTaskRunner()

Without lock:
1: Thread A test m_continue and get true then get ready into waiting state but not yet
2: Executor run stopProcessingWork() and notify_all
3: Thread A go into waiting state

Now Thread A will never be waken up